container/docker/install/install.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2019 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// The install package installs the docker container provider when imported
+package install
+
+import (
+	"time"
+
+	"github.com/google/cadvisor/container"
+	"github.com/google/cadvisor/container/docker"
+	"github.com/google/cadvisor/fs"
+	info "github.com/google/cadvisor/info/v1"
+	"github.com/google/cadvisor/watcher"
+	"golang.org/x/net/context"
+	"k8s.io/klog"
+)
+
+const dockerClientTimeout = 10 * time.Second
+
+func init() {
+	err := container.RegisterPlugin("docker", container.PluginFn{
+		InitializeContextFn: func(context *fs.Context) error {
+			docker.SetTimeout(dockerClientTimeout)
+			// Try to connect to docker indefinitely on startup.
```

Comment:
Is the Docker plugin the only one that retries the initialization because of the design of Docker itself? or were you planning on adding this connection retry logic for the other plugins at a later time? I'm just curious why the docker plugin is different from the others... thanks in advance for your answer!

Reply:
this is simply moving the existing behavior, I'll defer to the cadvisor maintainers on the reason for the current behavior

Reply:
Ah, cool. Sorry, didn't realize it was the existing behaviour...

Reply:
historical context: https://github.com/google/cadvisor/pull/1871

It was fixing a specific bug with docker. Not sure if the same bug could exist in the other runtimes...